x-pack/plugins/security/public/components/management/users/edit_user.js

```diff
@@ -96,6 +96,8 @@ export class EditUser extends Component {
     const { username } = this.state.user;
     if (username !== null && !username) {
       return 'Username is required';
+    } else if (!username.match(validUsernameRegex)) {
+      return 'Username contains invalid character(s)';
```

Comment:
I'm sorry to be such a pest... can we use the original error message here? I feel like it'd be more helpful to users trying to figure out what they did wrong:
> Username must begin with a letter or underscore and contain only letters, underscores, and numbers

Reply:
Lol, you're not being a pest, should have just gotten the original error message in the first place.